core/modules/ccontrol/ParseRunner.h

```diff
@@ -0,0 +1,122 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2012-2019 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_PARSER_PARSERUNNER_H
+#define LSST_QSERV_PARSER_PARSERUNNER_H
+
+
+// System headers
+#include <memory>
+#include <sstream>
+
+// Qserv headers
+#include "util/common.h"
+
+// Forward declarations
+namespace lsst {
+namespace qserv {
+namespace ccontrol {
+    class ParseListener;
+    class UserQuery;
+    class UserQueryResources;
+}
+namespace query {
+    class SelectStmt;
+}}} // End of forward declarations
+
+
+namespace lsst {
+namespace qserv {
+namespace ccontrol {
+
+
+/// ParseRunner drives the antlr4-based SQL parser.
+class ParseRunner {
+public:
+
+    /**
+     * @brief Construct a new Parse Runner object. Runs the parse.
+     *
+     * @param statement The sql statement to parse
+     *
+     * @throws parser::ParseException if the query can not be parsed.
+     */
+    ParseRunner(std::string const& statement);
+
+    /**
+     * @brief Construct a new Parse Runner object
+     *
+     * @param statement The sql statement to parse
+     * @param queryResources Resources that may be used to construct a UserQuery.
+     *
+     * @throws parser::ParseException if the query can not be parsed.
+     */
+    ParseRunner(std::string const& statement, std::shared_ptr<UserQueryResources> const& queryResources);
+
+    typedef std::shared_ptr<ParseRunner> Ptr;
+
+    /**
+     * @brief Convenience function to get a SelectStatement.
+     *
+     * This calls the ParseRunner constructor, so it may throw if the constructor throws.
+     *
+     * @param statement The sql statement to parse.
+     * @return std::shared_ptr<query::SelectStmt>
```

Comment:
I think you are supposed to say what it returns, not the type of the thing it returns (that is already obvious from declaration)

Reply:
Huh. It's just what my auto doxygen formatter writes for me, and I was lazy about writing an explanation. But you're right about it being redundant info, in C++.